website/versioned_docs/version-1.3.0/launcher.md

```diff
@@ -0,0 +1,116 @@
+---
+id: version-1.3.0-launcher-reference
+title: Launcher Reference
+sidebar_label: Built-in Launcher
+original_id: launcher-reference
+---
+
+The launcher is a lightweight artifact (~122K) that installs and configures
+bloop. The goal of the launcher is to relieve clients of the burden of
+installing and managing the lifetime of background build servers.
+
+The launcher implements the [Build Server Discovery protocol][server-discovery]
+which allows clients to establish a bsp connection and communicate with the
+server via stdin and stdout.
+
+## `launcher`
+
+#### Usage
+
+**Synopsis**: `launcher [JVM_OPTS...] [FLAGS...] BLOOP_VERSION`
+
+* `BLOOP_VERSION` must be a bloop version >= 1.2.0. This version is only
+  used in case an installation is tried.
+
+  > There is no guarantee that the server provided to the client will be of this version.
+
+* `JVM_OPTS` must be valid JVM arguments prefixed with `-J`.
+
+#### Flags
+
+<dl>
+<dt><code>--skip-bsp-connection</code> (type: <code>bool</code>)</dt>
+<dd><p>Do not attempt a bsp connection, but exit as soon as bloop has been installed and a server is running in the background.</p></dd>
+</dl>
+
+## Description
+
+The high-level logic steps of the launcher can be summarized as follows:
+
+1. If bloop is not detected in the `$PATH` or `$HOME/.bloop/`, install it.
+   * If the universal installation (that requires `curl` and `python`) fails,
+     then we attempt to resolve bloop and launch an embedded server with
+     a concrete bsp command. This operation is only atempted iff
+     `--skip-bsp-connection` is not used.
+2. Run a server in the background if it's absent.
+3. When a server is running:
+   * If `--skip-bsp-connection`: return
+   * Else establish a bsp connection with the server and forward any message
+     from the launcher stdin to the server and from the server to the launcher
+     stdout.
+
+### Reuse server in the background
+
+The launcher will always try to reuse a server already running or spawned by
+previous launcher executions to ensure that build servers are as hot and
+efficient as possible.
+
+### Use the launcher when off-line
+
+The launcher is designed to be off-line friendly, e.g. if after running the
+launcher there is no internet connection, the launcher will not need to hit
+internet again and will instead populate data from caches. The launcher will
+fail fatally in off-line scenarios if and only if the server is not installed
+from the start.
+
+## Usage
+
+### Run the launcher manually
+
+Clients can depend on the launcher and run it when they see fit. The flag
+`--skip-bsp-connection` is particularly useful for clients wishing to install
+and spawn a bloop version, but that do not wish to use BSP and instead prefer
+to detect manually the installed `bloop` version by the launcher and [run the
+CLI](cli-reference.md).
+
+#### Install the launcher
```

Comment:
is this a coursier command? maybe it's worth adding the complete example

Reply:
It's part of a coursier command but it's not supposed to be runnable. The original goal is that people realize which versions are published to Sonatype and which ones are only published to our personal bintray repository, if that makes sense!

Reply:
i meant in general, as a user - it's unclear how to get the launcher on my machine

Reply:
fair enough, I'll look into how I can improve on this tomorrow